optional_plugins/mail/avocado_mail_result/mail_result.py

```diff
@@ -0,0 +1,257 @@
+                    <p><strong>Job Finished At:</strong> {finishedtime}</p>
+                    <p><strong>Results:</strong></p>
+                    <ul>
+                        <li>PASS: {job.result.passed}</li>
+                        <li>ERROR: {job.result.errors}</li>
+                        <li>FAIL: {job.result.failed}</li>
+                        <li>SKIP: {job.result.skipped}</li>
+                        <li>WARN: {job.result.warned}</li>
+                        <li>INTERRUPT: {job.result.interrupted}</li>
+                        <li>CANCEL: {job.result.cancelled}</li>
+                    </ul>
+                    <p><strong>Test Summary:</strong></p>
+                    <pre>{test_summary}</pre>
+                </body>
+            </html>
+            """
+            msg = MIMEText(body, "html")
+            msg["Subject"] = (
+                f"{job.config.get('plugins.mail.header')} Job {job.unique_id} - Status: Job Completed"
+            )
+        else:
+            body = f"""
+            <html>
+                <body>
+                    <h2>Job Notification - Job {job.unique_id}</h2>
+                    <p><strong>Job Started At:</strong> {time_content}</p>
+                </body>
+            </html>
+            """
+            msg = MIMEText(body, "html")
+            msg["Subject"] = (
+                f"{job.config.get('plugins.mail.header')} Job {job.unique_id} - Status: Job Started"
+            )
+        msg["From"] = job.config.get("plugins.mail.sender")
+        msg["To"] = job.config.get("plugins.mail.recipient")
+        return msg
+
+    @staticmethod
+    def _send_email(smtp, sender, rcpt, msg):
+        try:
+            smtp.sendmail(sender, [rcpt], msg.as_string())
+            LOG_UI.info("EMAIL SENT TO: %s", rcpt)
+        except Exception as e:  # pylint: disable=W0703
+            LOG_UI.error(f"Failure to send email notification to {rcpt}: {e}")
+
+    @staticmethod
+    def _create_smtp_connection(server, port):
+        try:
+            smtp = smtplib.SMTP(server, port)
+            smtp.starttls()  # Enable TLS
+            return smtp
+        except Exception as e:  # pylint: disable=W0703
+            LOG_UI.error(f"Failed to establish SMTP connection to {server}:{port}: {e}")
+            return None
+
+    @staticmethod
+    def _read_results_file(results_path):
+        try:
+            with open(results_path, "r", encoding="utf-8") as file:
+                return json.load(file)
+        except FileNotFoundError:
+            LOG_UI.error("Test summary file not found at %s.", results_path)
+        except json.JSONDecodeError:
+            LOG_UI.error("Error decoding JSON from file %s.", results_path)
+        except Exception as e:
+            LOG_UI.error("Unexpected error while reading test summary: %s", str(e))
+        return None
+
+    @staticmethod
+    def _format_test_details(test, advanced=False):
+        if advanced:
```

Comment:
I would use [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) in this method. IMO, it will increase readability and maintainability, and it is a good practice for clean and robust code.